addons/binding/org.openhab.binding.homematic/OSGI-INF/HomematicThingHandlerFactory.xml

```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+    All rights reserved. This program and the accompanying materials
+    are made available under the terms of the Eclipse Public License v1.0
+    which accompanies this distribution, and is available at
+    http://www.eclipse.org/legal/epl-v10.html
+-->
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" immediate="true" name="org.openhab.binding.homematic.handler.HomematicThingHandlerFactory">
+   <implementation class="org.openhab.binding.homematic.handler.HomematicThingHandlerFactory"/>
+   <service>
+      <provide interface="org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory"/>
+   </service>
+
+   <reference bind="setTypeGenerator" cardinality="1..1" interface="org.openhab.binding.homematic.type.HomematicTypeGenerator" name="TypeGenerator" policy="static" unbind="unsetTypeGenerator"/>
+   <reference bind="setItemChannelLinkRegistry" cardinality="1..1" interface="org.eclipse.smarthome.core.thing.link.ItemChannelLinkRegistry" name="ItemChannelLinkRegistry" policy="static" unbind="unsetItemChannelLinkRegistry"/>
```

Comment:
you should not depend on the ItemChannelLinkRegistry - see my comments below in the code.